inlong-sort/sort-formats/format-rowdata/format-rowdata-json/src/main/java/org/apache/inlong/sort/formats/json/JsonRowDataDeserializationSchema.java

```diff
@@ -104,13 +103,20 @@ public static Builder builder(RowType rowType, TypeInformation<RowData> resultTy
     }
 
     @Override
-    public RowData deserializeInternal(@Nullable byte[] message) throws IOException {
+    public RowData deserializeInternal(@Nullable byte[] message) throws Exception {
         if (message == null) {
             return null;
         }
 
         String jsonStr = new String(message, charset);
-        return (RowData) runtimeConverter.convert(jsonStr);
+        RowData rowData = null;
+        try {
+            rowData = (RowData) runtimeConverter.convert(jsonStr);
+        } catch (Throwable t) {
+            failureHandler.onParsingMsgFailure(message, new RuntimeException(
```

Comment:
For readability, `jsonStr` is better than `message`

Reply:
onParsingMsgFailure's parameter is byte type. josnStr is string